src/frontends/paddle/src/op/unstack.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs unstack(const NodeContext& node) {
+    auto data = node.get_input("X");
+    auto dim = node.get_attribute<int32_t>("axis");
+    if (dim < 0) {
+        dim = dim + static_cast<int32_t>(data.get_partial_shape().rank().get_length());
+    }
+    auto axis = default_opset::Constant::create(element::i32, {}, {dim});
+    auto shape = data.get_shape();
```

Comment:
```suggestion
    auto input_shape = data.get_partial_shape();
    PADDLE_OP_CHECK(node,
                        input_shape.rank().is_static(),
                        "rank of input data should be static");
    auto shape = input_shape.get_shape();
```
seems better if we add a static check here